### ISSUE TITLE
embedding video failed when using `embed-resources: true` or `self-contained: true` in the YAML.

### ISSUE THREAD

**Issue by elgabbas** [quarto-dev/quarto-cli#3310]
### Bug description

Hello,

I am trying to embed a youtube video into a Quarto document (v1.2.269 on windows 11 pro).

However, I noticed that the video embedding failed when setting either `embed-resources: true` or `self-contained: true` in the YAML.

The same occurred using the `{{< video LINK >}}` shortcode or `iframe`. `iframe` works only after setting `data-external="1"`; e.g. `<iframe data-external="1" src="LINK"></iframe>`

****

The following example does not work
```
---
title: "Quarto"
format:
  html:
    toc: true
    self-contained: true
---
{{< video https://www.youtube.com/embed/wo9vZccmqwc >}}
```

```
---
title: "Quarto"
format:
  html:
    toc: true
    embed-resources: true
---
{{< video https://www.youtube.com/embed/wo9vZccmqwc >}}
```
****

While this works:
```
---
title: "Quarto"
format:
  html:
    toc: true
---
{{< video https://www.youtube.com/embed/wo9vZccmqwc >}}
```
****
Is this an expected behaviour?

Thanks, Ahmed

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [ ] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
This is now fixed on `main`, and should be available on the next 1.3 prerelease build. Thanks!

**Comment by elgabbas** [comment]
Great.... Thanks....
